src/popup/modules/InitQrCode.js

```diff
@@ -85,3 +85,38 @@ export const initiationProcess = Promise.all([qrCreatorInit, userInterfaceInit])
 }).catch((error) => {
     Logger.logError(error);
 });
+
+/**
+ * Queries Tabs API for 20 times with a delay of 300ms until the tabs have a defined url.
+ * Will reject if url is not defined.
+ * @returns {Promise}
+ */
+async function getsTabWithValidUrl() {
+    let retryCount = 0;
+    const maxRetries = 20;
+    const delay = 300;
+
+    try {
+        return await retryPromise(async () => {
+            if (retryCount === maxRetries) {
+                throw new Error("No tabs with url.");
+            }
+
+            try {
+                retryCount++;
+
+                const tabs = await queryBrowserTabs();
+
+                if (tabs && tabs[0].url) {
+                    return tabs;
+                } else {
+                    throw new Error("Url not found.");
+                }
+            } catch (error) {
+                throw error;
```

Comment:
This catches and directly throws the error again?? I guess this is not needed. So we can have one less try block here… Same below. Thrown errors bubble "through" until they are catched. So there is no need to catch them, just to throw them directly thereafter. (They work like exceptions.)